packages/apollo-gateway/src/__tests__/queryPlanCucumber.test.ts

```diff
@@ -0,0 +1,138 @@
+import { buildQueryPlan, buildOperationContext } from '../buildQueryPlan';
+
+const buildQueryPlanFeature = loadFeature(
+  './packages/apollo-gateway/src/__tests__/build-query-plan.feature'
+);
+
+
+const features = [
+  buildQueryPlanFeature
+];
+
+function buildLocalService(modules: GraphQLSchemaModule[]) {
+  const schema = buildFederatedSchema(modules);
+  return new LocalGraphQLDataSource(schema);
+}
+
+features.forEach((feature) => {
+  defineFeature(feature, (test) => {
+    feature.scenarios.forEach((scenario) => {
+      test(scenario.title, async ({ given, when, then }) => {
+        let query: DocumentNode;
+        let schema: GraphQLSchema;
+        let errors: GraphQLError[];
+        let queryPlan: QueryPlan;
+        let options: any = {};
+
+        const serviceMap = Object.fromEntries(
+          ['accounts', 'product', 'inventory', 'reviews', 'books', 'documents'].map(
+            serviceName => {
+              return [
+                serviceName,
+                buildLocalService([
+                  require(path.join(
+                    __dirname,
+                    '__fixtures__/schemas',
+                    serviceName,
+                  )),
+                ]),
+              ] as [string, LocalGraphQLDataSource];
+            },
+          ),
+        );
+
+        ({ schema, errors } = composeServices(
+          Object.entries(serviceMap).map(([serviceName, service]) => ({
+            name: serviceName,
+            typeDefs: normalizeTypeDefs(service.sdl()),
+          })),
+        ));
+
+        if (errors && errors.length > 0) {
+          throw new GraphQLSchemaValidationError(errors);
+        }
+
+        const givenQuery = () => {
+          given(/^query$/im, (operation) => {
+            query = gql(operation)
+          })
+        }
+
+        const whenUsingAutoFragmentization = () => {
+          when(/using autofragmentization/i, () => {
+            options = { autoFragmentization: true };
+          })
+        }
+
+        const thenQueryPlanShouldBe = () => {
+          then(/^query plan$/i, (expectedQueryPlan) => {
+            queryPlan = buildQueryPlan(
+              buildOperationContext(schema, query, undefined),
+              options
+            );
+
+            const serializedPlan = JSON.parse(JSON.stringify(queryPlan, serializeQueryPlanNode));
+            const parsedExpectedPlan = JSON.parse(expectedQueryPlan);
+
+            expect(serializedPlan).toEqual(parsedExpectedPlan);
+          })
+        }
+
+        // step over each defined step in the .feature and execute the correct
+        // matching step fn defined above
+        scenario.steps.forEach(({ stepText }) => {
+          if (/^query$/i.test(stepText)) givenQuery();
+          else if (/using autofragmentization/i.test(stepText)) whenUsingAutoFragmentization();
+          else if (/^query plan$/i.test(stepText)) thenQueryPlanShouldBe();
+          else throw new Error('Invalid steps in .feature file');
```

Comment:
Should we include the name of _what_ feature file in this error to help debugging?

Reply:
I don't have an _easy_ way to make this flexible and print the name of the file where this came from for test suites with multiple feature files. I can do this right now, since there's only one file in this suite, but for the integration tests, I'd need to do a little more (still manageable) work

Reply:
Ok! Not necessary to do it right now.  There's only one file that it could be right now, as you noted!